client/lib/i18n-utils/switch-locale.js

```diff
@@ -547,15 +547,33 @@ function loadCSS( cssUrl, currentLink ) {
 }
 
 /**
- * Add translations.
+ * Translation data batch strore.
+ *
+ * @type {Array}
+ */
+const _translationsBatch = [];
+
+/**
+ * A throttle wrapper around i18n.addTranslations.
+ *
+ * @param {Object} userTranslations User translations data that will override chunk translations
+ */
+const _addTranslationsBatch = throttle( function ( userTranslations ) {
+	i18n.addTranslations( Object.assign( {}, ..._translationsBatch.splice( 0 ), userTranslations ) );
```

Comment:
This should clear `_translationsBatch`, otherwise the next throttled call will re-add them.

Reply:
`Array.splice` is emptying the array, no?

Reply:
Yep, you are right. I still confuse `slice` and `splice` 🤦🏼